### PR TITLE
Manage new way to get error Output

### PR DESCRIPTION
**Pull request by ddubuc**
Issue : https://github.com/aneoconsulting/ArmoniK/issues/276

### CODE REVIEW

File: Common/ArmoniK.DevelopmentKit.Common/Submitter/BaseClientSubmitter.cs [hunk 1]
```diff
@@ -124,10 +124,52 @@ public void HealthCheckResult(string taskId)
...
+    }
+
+    /// <summary>
+    /// Returns the status of the task
+    /// </summary>
+    /// <param name="taskId">The taskId of the task</param>
+    /// <returns></returns>
+    public TaskStatus GetTaskStatus(string taskId)
+    {
+      var status = GetTaskStatues(taskId).First();
```
Comment (aneojgurhem) [inline Common/ArmoniK.DevelopmentKit.Common/Submitter/BaseClientSubmitter.cs:137]: Single ? Tu es censé en recevoir qu'un seul

File: Common/EndToEnd.Tests/Tests/CheckUnifiedApi/SimpleUnfiedAPITest.cs [hunk 2]
```diff
@@ -25,7 +25,9 @@
 using System.Collections.Generic;
 using System.Linq;
 
+using ArmoniK.DevelopmentKit.Client.Exceptions;
 using ArmoniK.DevelopmentKit.Common;
+using ArmoniK.DevelopmentKit.Common.Exceptions;
```
Comment (aneojgurhem) [inline Common/EndToEnd.Tests/Tests/CheckUnifiedApi/SimpleUnfiedAPITest.cs:30]: il y a que `GridServerException` d'ajouté donc pourquoi ces 2 imports ?

File: UnifiedApi/Client/Exceptions/ServiceInvocationException.cs [hunk 3]
```diff
@@ -25,30 +25,66 @@
...
+      message_        = message;
+      StatusCode = statusCode;
+      }
 
-    public ServiceInvocationException(Exception e) : base(e.Message,
-                                                          e) => message_ = $"{message_} with InnerException {e.GetType()} message : {e.Message}";
+    /// <summary>
+    /// The overriden constructor to accept inner Exception as parameters
+    /// </summary>
+    /// <param name="e">Th previous exception</param>
```
Comment (aneojgurhem) [inline UnifiedApi/Client/Exceptions/ServiceInvocationException.cs:63]: manque un e

File: UnifiedApi/Client/Exceptions/ServiceInvocationException.cs [hunk 4]
```diff
@@ -25,30 +25,66 @@
...
+    }
 
-    public ServiceInvocationException(string message, ArgumentException e) : base(message,
-                                                                                  e)
-      => message_ = message;
+    /// <summary>
+    /// The overriden constructor to acception inner exception and message as parameters
+    /// </summary>
+    /// <param name="message">the message to set in the exception</param>
+    /// <param name="e">the previous exception generated by failure</param>
```
Comment (aneojgurhem) [inline UnifiedApi/Client/Exceptions/ServiceInvocationException.cs:76]: des majuscules au debut pour etre cohérent

File: UnifiedApi/Client/Services/Submitter/Service.cs [hunk 5]
```diff
@@ -79,6 +76,37 @@ public class ServiceResult
 
     private Dictionary<string, Task> TaskWarehouse { get; set; } = new();
 
+
+    // *** you need some mechanism to map types to fields
```
Comment (aneojgurhem) [inline UnifiedApi/Client/Services/Submitter/Service.cs:80]: Yes ! This is called "using the existing types"

File: UnifiedApi/Client/Services/Submitter/Service.cs [hunk 6]
```diff
@@ -166,11 +194,35 @@ public ServiceResult Execute(string methodName, byte[] dataArg)
...
+      var   taskId = "not-TaskId";
+      object[] result;
 
-      var result = ProtoSerializer.DeSerializeMessageObjectArray(SessionService.GetResult(taskId));
+      try
+      {
+        taskId = SessionService.SubmitTask(dataSynapsePayload.Serialize());
 
-      return new ServiceResult()
+        result = ProtoSerializer.DeSerializeMessageObjectArray(SessionService.GetResult(taskId));
```
Comment (aneojgurhem) [inline UnifiedApi/Client/Services/Submitter/Service.cs:204]: perf ?

**Review by aneojgurhem — CHANGES_REQUESTED** [review]

File: Common/ArmoniK.DevelopmentKit.Common/Submitter/BaseClientSubmitter.cs (hunk 1 above)
Comment (ddubuc) [reply Common/ArmoniK.DevelopmentKit.Common/Submitter/BaseClientSubmitter.cs:137]: yes je corrige

File: Common/EndToEnd.Tests/Tests/CheckUnifiedApi/SimpleUnfiedAPITest.cs (hunk 2 above)
Comment (ddubuc) [reply Common/EndToEnd.Tests/Tests/CheckUnifiedApi/SimpleUnfiedAPITest.cs:30]: Nettoyé

File: UnifiedApi/Client/Exceptions/ServiceInvocationException.cs (hunk 4 above)
Comment (ddubuc) [reply UnifiedApi/Client/Exceptions/ServiceInvocationException.cs:76]: Done

File: UnifiedApi/Client/Services/Submitter/Service.cs (hunk 5 above)
Comment (ddubuc) [reply UnifiedApi/Client/Services/Submitter/Service.cs:80]: Idea here is to merge status of task and status of result for HandleError.

Comment (aneojgurhem) [reply UnifiedApi/Client/Services/Submitter/Service.cs:80]: That will not be the case in protos

**Review by aneojgurhem — APPROVED** [review]

File: UnifiedApi/Client/Services/Submitter/Service.cs (hunk 6 above)
Comment (ddubuc) [reply UnifiedApi/Client/Services/Submitter/Service.cs:204]: Even if the protoSerilizer is fast and optionel. Only if Result Type is not byte[] in the method signature

**Pull request merged**